python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -20,179 +20,197 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 import numpy as np
-from scarlet.source import PointSource, ExtendedSource, SourceInitError
-from scarlet.component import BlendFlag, Component
+from scarlet.source import PointSource, ExtendedSource, MultiComponentSource
 
 import lsst.afw.image as afwImage
 from lsst.afw.geom import SpanSet
 from lsst.geom import Point2I
+import lsst.log
 import lsst.afw.detection as afwDet
 
-__all__ = ["LsstSource", "LsstHistory"]
-
-
-class BlankSource(Component):
-    """Placeholder for a source that cannot be initialized
-
-    In DM-19790 this class will be removed when skipped sources are
-    not deblended at all.
+__all__ = ["init_source", "morphToHeavy", "modelToHeavy"]
+
+logger = lsst.log.Log.getLogger("meas.deblender.deblend")
+
+
+def init_source(frame, peak, observation, bbox,
+                symmetric=False, monotonic=True,
+                thresh=5, components=1):
+    """Initialize a Source
+
+    The user can specify the number of desired components
+    for the modeled source. If scarlet cannot initialize a
+    model with the desired number of components it continues
+    to attempt initialization of one fewer component until
+    it finds a model that can be initialized.
+
+    It is possible that scarlet will be unable to initialize a
+    source with the desired number of components, for example
+    a two component source might have degenerate components,
+    a single component source might not have enough signal in
+    the joint coadd (all bands combined together into
+    single signal-to-noise weighted image for initialization)
+    to initialize, and a true spurious detection will not have
+    enough signal to initialize as a point source.
+    If all of the models fail, including a `PointSource` model,
+    then this source is skipped.
+
+    Parameters
+    ----------
+    frame: `LsstFrame`
```

Comment:
Need a space before and after the colon here and in most of the added parameter docstrings.

Sadly I haven't been able to convince PyCharm to do this automatically.

Reply:
This is an odd style guide requirement, as DM typically doesn't adhere to the spaces around operators rule. Numpydocs doesn't require this to compile properly, so this is just a waste of time (not your fault, I'm just venting). But I fixed it throughout the package, since I never use a space on both sides of the colon (because why would a sane person!).